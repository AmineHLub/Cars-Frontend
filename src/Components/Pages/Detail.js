/* eslint-disable jsx-a11y/control-has-associated-label */
import { MdArrowBack } from 'react-icons/md';
import { GrAddCircle } from 'react-icons/gr';

const Detail = ({ selectedCar, setSelectedCar }) => (
  <div className="detail-page-container">
    <div className="selected-car-img-container">
      <img src={selectedCar.image} alt="selected-car" />
    </div>
    <div className="selected-car-info-container d-flex">
      <div className="header-car-info">
        <h2>{selectedCar.name}</h2>
        <p>{selectedCar.description}</p>
      </div>
      <div className="all-fees-container">
        <div className="fees-container d-flex">
          <p>
            One day reservation:
          </p>
          <p>
            {selectedCar.reservation_fee}
            {' '}
            $
          </p>
        </div>
        <div className="fees-container d-flex">
          <p>
            Purchase Price:
          </p>
          <p>
            {selectedCar.reservation_fee * 678}
            {' '}
            $
          </p>
        </div>
      </div>
      <button
        type="button"
        className="add-reservation-btn d-flex"
        onClick={() => setSelectedCar('')}
      >
        <GrAddCircle />
        <p>Reserve</p>
      </button>
    </div>
    <button
      type="button"
      className="go-back-btn"
      onClick={() => setSelectedCar('')}
    >
      <MdArrowBack />
    </button>
  </div>
);
export default Detail;
