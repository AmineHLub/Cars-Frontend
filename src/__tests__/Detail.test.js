import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Detail from '../components/Pages/Detail';

const setPending = [];
const pendingReservations = [];
const selectedCar = {
  image: 'test.png',
  name: 'test',
  description: 'test description',
  reservation_fee: 100,
};
const setSelectedCar = [];

describe('Testing Detail component', () => {
  test('renders a correct data', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Detail
            selectedCar={selectedCar}
            setSelectedCar={setSelectedCar}
            setPending={setPending}
            pendingReservations={pendingReservations}
          />
        </Router>
      </Provider>,
    );
    expect(getByTestId('header-car-info')).toBeInTheDocument();
    expect(getByTestId('all-fees-container')).toBeInTheDocument();
    expect(getByTestId('reservation-button')).toBeInTheDocument();
    expect(getByTestId('selected-car-img-container')).toBeInTheDocument();
  });

  test('renders as expected', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Detail
            selectedCar={selectedCar}
            setSelectedCar={setSelectedCar}
            setPending={setPending}
            pendingReservations={pendingReservations}
          />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
