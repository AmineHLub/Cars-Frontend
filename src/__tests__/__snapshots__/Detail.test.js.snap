// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Detail component renders as expected 1`] = `
<div
  className="detail-page-container"
>
  <div
    className="selected-car-img-container"
    data-testid="selected-car-img-container"
  >
    <img
      alt="selected-car"
      src="test.png"
    />
  </div>
  <div
    className="selected-car-info-container d-flex"
  >
    <div
      className="header-car-info"
      data-testid="header-car-info"
    >
      <h2>
        test
      </h2>
      <p>
        test description
      </p>
    </div>
    <div
      className="all-fees-container"
      data-testid="all-fees-container"
    >
      <div
        className="fees-container d-flex"
      >
        <p>
          One day reservation:
        </p>
        <p>
          100
           
          $
        </p>
      </div>
      <div
        className="fees-container d-flex"
      >
        <p>
          Purchase Price:
        </p>
        <p>
          67800
           
          $
        </p>
      </div>
    </div>
    <button
      className="add-reservation-btn d-flex"
      data-testid="reservation-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,22 C17.5228475,22 22,17.5228475 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,17.5228475 6.4771525,22 12,22 Z M12,18 L12,6 M6,12 L18,12"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
      <p>
        Reserve
      </p>
    </button>
  </div>
  <button
    className="go-back-btn"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
  </button>
</div>
`;
